projects/storefrontlib/src/cms-structure/page/slot/page-slot.component.ts

```diff
@@ -14,145 +14,184 @@ import {
   CmsService,
   ContentSlotComponentData,
   ContentSlotData,
-  DeferLoadingStrategy,
   DynamicAttributeService,
 } from '@spartacus/core';
 import { BehaviorSubject, Observable, Subscription } from 'rxjs';
 import { distinctUntilChanged, map, switchMap, tap } from 'rxjs/operators';
 import { IntersectionOptions } from '../../../layout/loading/intersection.model';
 
+/**
+ * The `PageSlotComponent` is used to render the CMS page slot and it's components.
+ *
+ * The Page slot host element will be supplemented with css classes so that the layout
+ * can be fully controlled by customers:
+ * - The page slot _position_ is added as a css class by default.
+ * - The `cx-pending` is added for as long as the slot hasn't start loading.
+ * - The `page-fold` style class is added for the page slot which is configured as the page fold.
+ */
 @Component({
   selector: 'cx-page-slot,[cx-page-slot]',
   templateUrl: './page-slot.component.html',
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class PageSlotComponent implements OnInit, OnDestroy {
   /**
-   * The position is used to find the CMS page slot (and optional outlet)
-   * that is rendered in the PageSlotComponent. Furthermore, the position
-   * is added as a CSS class name to the host element.
+   * The position represents the unique key for a page slot on a single page, but can
+   * be reused cross pages.
+   *
+   * The position is used to find the CMS components for the page slot. It is also
+   * added as an additional CSS class so that layoutt can be applied.
    */
-  @Input()
-  set position(position: string) {
-    this.position$.next(position);
-    this.renderer.addClass(this.hostElement.nativeElement, position);
+  @Input() set position(value: string) {
+    this.position$.next(value);
   }
   get position(): string {
-    return this.position$.value;
+    return this.position$?.value;
```

Comment:
Do we need `?` here?

Reply:
I thought yes, but indeed it's not needed as the BehaviourSubject is initialised explicitly.